ADR/####-slsa-source-provenance-verification.md

```diff
@@ -0,0 +1,133 @@
+
+However, Konflux's build-provenance-centered architecture prompts a question: **How should we verify and incorporate source provenance into our existing trust model?**
+
+Some different approaches are possible:
+
+1. **Direct reference**: Include source provenance attestations directly in build provenance
+2. **Parallel verification**: Verify source provenance independently alongside build provenance
+3. **Chained attestations**: Create build provenance attestations about source provenance attestations
+
+The challenge is that the source track's storage model (git notes in repositories) differs significantly from our build artifact storage model (OCI registries), making it non-obvious how to integrate source provenance verification into our existing Conforma policies and attestation chains.
+
+### References
+
+- [SLSA Source Track Requirements](https://slsa.dev/spec/draft/source-requirements) - Source provenance specification
+- [SLSA Verification Summary Attestations](https://slsa.dev/spec/draft/verification_summary) - VSA format specification
+- [ADR 48: Movable Build Tests](0048-movable-build-tests.md) - Pattern for independent verification tasks
+- [ADR 49: VSA Support](0049-vsa-support.md) - Trusting VSAs for externally-built artifacts
+- [ADR 36: Trusted Artifacts](0036-trusted-artifacts.md) - OCI storage patterns for artifacts
+- [ADR 20: Source Retention](0020-source-retention.md) - Existing source requirements
+
+## Decision
+
+We will verify SLSA source provenance through a **chained attestation** approach that leverages the movable test pattern from [ADR 48](0048-movable-build-tests.md).
+
+### Verification Workflow
+
+1. **Source VSA generation** (external to Konflux):
+   - Source repository maintainers configure automation (GitHub Actions, GitLab CI/CD pipelines, etc.) to generate source VSAs
+   - These VSAs are stored as git notes in the source repository, contemporaneously with commits
+   - VSAs attest to source control practices (e.g., two-person review, branch protection, audit logs)
+
+2. **Source provenance verification task** (in Konflux pipeline):
+   - A Konflux pipeline task reads the source VSA from git notes for the cloned commit
+   - The task accepts a public key parameter (provided via IntegrationTestScenario configuration) and verifies the source VSA signature against that key
+   - The task verifies the VSA contents (format, subject matching the commit, etc.)
+   - The task pushes the source VSA to the OCI registry as a separate artifact
+   - The task exposes the pullspec of the OCI representation of the VSA (digest reference) in its task results using `*_ARTIFACT_OUTPUTS` format
+   - The task records the public key used for verification in its results
```

Comment:
Nitpick: does the task need to return it as a result if it's already recorded in parameters? It's unforgeable in the attestation either way

One can never be too careful about returning too many bytes from Tekton Tasks